reducers/heroes.js

```diff
@@ -0,0 +1,18 @@
+import { heroes } from 'dotaconstants';
+import { API_HOST } from './../yasp.config';
+
+// lazy evaluation
+// add API_HOST to hero img path so components
+// don't need to know API_HOST
+const transformHeroImagePath = (heroes) => {
+  const heroList = {};
+  Object.keys(heroes).forEach((key) => {
+    heroList[key] = { ...heroes[key] };
+    heroList[key].img = API_HOST + heroList[key].img;
+  });
+
```

Comment:
So a selector is just something which retrieves something from your state. It can be a derived thing (like the getVisibleTodos in the redux docs), but it can also just be a piece of your state. So the pattern I am trying to establish here is that in mapStateToProps, we never have to do something like:
```js
const mapStateToProps = state => ({
  something: state.deep.in.your.tree.something,
  somethingElse: state.other.place.thats.deep.somethingElse,
});
```
This is coupling the connected component to the state of your shape. Maybe someday, you want to flatten your state, or you want to refactor your state to be normalized (more like the reducers in gotPlayer/match.js or any of those). So the pattern would be to encapsulate that state shape knowledge in selectors located in the reducers. For example:
```js
// reducer.js

const initialState = {
  reducer: {
    prop1: false,
    prop2: true,
    list: [],
  }
};

// ...your reducer code goes here...

export const reducer = {
  getReducer: state => state.yaspReducer.reducer,
  getProp1: state => getReducer.getReducer(state).prop1,
  getProp2: state => getReducer.getReducer(state).prop2,
  getList: state => getReducer.getReducer(state).list,
};

// connectedComponent.js
import { reducer } from '../../reducers';
import Component from './Component';
import { connect } from 'react-redux';

const mapStateToProps = state => ({
  prop1: reducer.getProp1(state),
  list: reducer.getList(state),
});

export default connect(mapStateToProps)(Component);
```

So in this form, your component will never know what the state shape is, it will just be using essentially getters to access the state (passing in the state as a param to the getter). So instead of keeping presentational components unaware of the state shape, but coupling your connected components to the state shape (so if you refactor you have to change all connected components as well or you could get mysterious and hard to diagnose errors).